neo/core/group.py

```diff
@@ -48,13 +58,82 @@ def __init__(self, objects=None, name=None, description=None, file_origin=None,
                  allowed_types=None, **annotations):
         super().__init__(name=name, description=description,
                          file_origin=file_origin, **annotations)
+
+        # note that we create the ObjectLists here _without_ a parent argument
+        # since objects do not have a reference to the group(s)
+        # they are contained in.
+        self._analogsignals = ObjectList(AnalogSignal)
+        self._irregularlysampledsignals = ObjectList(IrregularlySampledSignal)
+        self._spiketrains = SpikeTrainList()
+        self._events = ObjectList(Event)
+        self._epochs = ObjectList(Epoch)
+        self._channelviews = ObjectList(ChannelView)
+        self._imagesequences = ObjectList(ImageSequence)
+        self._segments = ObjectList(Segment)  # to remove?
```

Comment:
I vote for the removal of segments here to simplify the neo structure.

Reply:
I am OK with this. This can be done outside the PR no ?